sar/src/main/java/org/jboss/as/service/ParsedServiceDeploymentProcessor.java

```diff
@@ -241,7 +245,12 @@ private ObjectName createDependencyObjectName(final String dependencyName) throw
             paramTypes.add(attributeTypeValue);
             paramValues.add(new ImmediateValue<Object>(newValue(attributeTypeValue, parameter.getValue())));
         }
-        final Value<Method> methodValue = new ImmediateValue<Method>(ReflectionUtils.getMethod(mBeanClassHierarchy, methodName, paramTypes.toArray(new Class<?>[0]), true));
+        final Value<Method> methodValue = new InjectedBeanMethodValue(Values.injectedValue(), new InjectedBeanMethodValue.MethodFinder() {
+            @Override
+            public Method find(Class<?> clazz) {
+                return ReflectionUtils.getMethod(clazz, methodName, paramTypes.toArray(new Class<?>[0]));
+            }
+        });
         return cached(new MethodValue<Object>(methodValue, Values.injectedValue(), paramValues));
     }
 
```

Comment:
After these changes there are a few private methods in this class where the 'mbeanClassHierarchy' param is no longer used and should be removed.